python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -389,17 +463,20 @@ def launch_scheduled_runs_for_schedule_iterator(
                 start_timestamp_utc,
                 latest_tick.timestamp,
                 instigator_data.last_iteration_timestamp or 0.0,
+                in_memory_last_iteration_timestamp or 0.0,
```

Comment:
I also added this to reduce the likelihood that a cron string changing can mess things up, even if we increase the interval at which we write timestamps to the DB - now to run into trouble you have to be within that larger window AND the daemon has to restart

Reply:
went into it in the ScheduleIterationTimes constructor